tests/interpolate/interp1d_test.py

```diff
@@ -93,6 +93,24 @@ def test_close():
     assert sc.allclose(da.data, out.data, rtol=sc.scalar(1e-3))
 
 
+def test_fail_multidim_mask():
+    da = make_array()
+    da.masks['mask'] = da.data != da.data
+    with pytest.raises(sc.DimensionError):
+        interp1d(da, 'xx')
+
+
+def test_masked():
+    x = sc.linspace(dim='xx', start=0.0, stop=3.0, num=20, unit='rad')
+    da = sc.DataArray(sc.sin(x), coords={'xx': x})
+    da.masks['mask'] = da.data > sc.scalar(0.9)
+    result = interp1d(da, 'xx', kind='cubic')(da.coords['xx'])
+    assert sc.allclose(result.data, da.data, rtol=sc.scalar(3e-3))
+    da.masks['mask'] = da.data > sc.scalar(0.8)
+    result = interp1d(da, 'xx', kind='cubic')(da.coords['xx'])
```

Comment:
Not sure `'cubic'` is a good choice here. Cubic interpolations can swing above and below the data points quite far. So this may not be the most reliable choice for a test.

Reply:
I had tried this one and confirmed visually. Do you have a better suggestion?

Reply:
A linear interpolation would be more predictable (needs only the two points at the ends of the interval you are looking at.)
I don't know if this is a real issue. The test might break if `interp1d` changes. But it shouldn't...?

Reply:
Ok